### PR TITLE
I FUCKING HATE CONSTANT RIOTING

### LINKED ISSUE #1030: Invisible walls in mining ruins
#### Description of issue

Screenshot (135)

Was watching this ghost try and explore this area, but even after deleting the airlock and blast doors he couldn't go through (he already opened it with a crowbar, but there was an invisible wall).

#### Difference between expected and actual behavior

Open floor should be open, but mime magic blocks you.

#### Steps to reproduce

Spawn in this structure and mine to it as D-class
Try to walk in

#### Length of time in which bug has been known to occur

I haven't seen this before so IDK

#### Client version, Server revision & Game ID

Client Version: 514
Server Revision: Revision Unknown
Game ID: cnc-c7DD
Current map: Foundation Site 53

#### Issue bingo
- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [ ] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/foundation-19/foundation-19/issues)

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

D-block balance changes

Fixes #1030.

## Changelog

:cl:
balance: BODA dispenser was changed to a Water Dispenser, which does NOT dispense glass.
imageadd: BODA dispenser sprite altered.
balance: The Class-D seed servitor charges money.
balance: Disarming has been changed to be less buggy, less random, and less OP.
balance: The D-block built-in medbay has upped security. It should now be harder for D-class rioters to steal the dispenser.
del: Rogue maintenance drone event has been removed.
code: General improvements.
/:cl:

### CODE REVIEW

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -1040,9 +1040,9 @@
 	icon_deny = "nutri_generic-deny"
 
 /obj/machinery/vending/hydroseeds
-	name = "\improper MegaSeed Servitor"
-	desc = "When you need seeds fast!"
-	product_slogans = "THIS'S WHERE TH' SEEDS LIVE! GIT YOU SOME!;Hands down the best seed selection this half of the galaxy!;Also certain mushroom varieties available, more for experts! Get certified today!"
+	name = "\improper MegaSneed Servitor"
+	desc = "When you need seeds fast! Formerly Chuck's."
```
Comment (TheDarkElites) [inline code/game/machinery/vending.dm:1045]: 10/10

Comment (aidan-bowman) [reply code/game/machinery/vending.dm:1045]: :)

File: code/_helpers/unsorted.dm [hunk 2]
```diff
@@ -237,7 +237,7 @@ Turf and target are seperate in case you want to teleport some distance from a t
 			line+=locate(px,py,M.z)
 	return line
 
-#define LOCATE_COORDS(X, Y, Z) locate(between(1, X, world.maxx), between(1, Y, world.maxy), Z)
+#define LOCATE_COORDS(X, Y, Z) locate(Clamp(X, 1, world.maxx), Clamp(Y, 1, world.maxy), Z)
```
Comment (EgorDinamit) [inline code/_helpers/unsorted.dm]: clamp() is default Byond expression, use it instead.
```suggestion
#define LOCATE_COORDS(X, Y, Z) locate(clamp(X, 1, world.maxx), clamp(Y, 1, world.maxy), Z)
```

Comment (aidan-bowman) [reply code/_helpers/unsorted.dm]: > clamp() is default Byond expression, use it instead.

goddamn it, welp time to regex search and replace

**Comment by Tsurupeta** [comment]
Revert clamp commit and make them separate PR, out of scope

**Comment by Tsurupeta** [comment]
Actually it's better to just drop it I think

**Comment by aidan-bowman** [comment]
> Revert clamp commit and make them separate PR, out of scope

krill issue

**Comment by aidan-bowman** [comment]
> Actually it's better to just drop it I think

can't just make a new PR? i mean it's technically an optimization, plus it makes code more readable

**Comment by Tsurupeta** [comment]
I meant drop those commits instead if reverting them

**Comment by Vizoplez** [comment]
This entirely removes skills...?

No?

**Comment by Tsurupeta** [comment]
99% sure it was an unticked file

File: code/modules/species/species.dm [hunk 3]
```diff
@@ -647,29 +647,28 @@ The slots that you can use are found in items_clothing.dm and are the inventory
...
-	if(target.a_intent == I_HELP)
-		state_mod -= 30
+	var/state_mod = attacker.melee_accuracy_mods() - target.melee_accuracy_mods() + ((target.a_intent == I_HELP) ? 15 : 0)
+	var/stim_mod = attacker.chem_effects[CE_SPEEDBOOST] - attacker.chem_effects[CE_SLOWDOWN] + target.chem_effects[CE_SEDATE] - target.chem_effects[CE_STIMULANT]
+	var/push_mod = Clamp((1 + attacker.get_skill_difference(SKILL_COMBAT, target)), 1, 3)
+
 	//Handle unintended consequences
 	for(var/obj/item/I in holding)
-		var/hurt_prob = max(holding[I] - 2*skill_mod + state_mod, 0)
+		var/hurt_prob = Clamp(holding[I] - (2 * skill_mod) + state_mod, 0, 75)
```
Comment (EgorDinamit) [inline code/modules/species/species.dm]:
```suggestion
		var/hurt_prob = clamp(holding[I] - (2 * skill_mod) + state_mod, 0, 75)
```

**Review by EgorDinamit — COMMENTED** [review]
Any reason you moved chem dispenser from literal chemistry to middle of nowhere in caves?

**Comment by aidan-bowman** [comment]
> Any reason you moved chem dispenser from literal chemistry to middle of nowhere in caves?

was recommended to me by bobinator

**Comment by aidan-bowman** [comment]
well fuck, how do i resolve a map conflict like this

**Comment by aidan-bowman** [comment]
fuck it, gun shit is outside the scope of this PR

**Comment by aidan-bowman** [comment]
that should be good for now, please speedmerge so i don't have to resolve another map conflict

File: code/game/objects/items/weapons/tools/crowbar.dm [hunk 4]
```diff
@@ -70,3 +70,8 @@
 			return FALSE
 
 	return TRUE
+
+/obj/item/crowbar/makeshift
+	name = "improvised Crowbar"
```
Comment (EgorDinamit) [inline code/game/objects/items/weapons/tools/crowbar.dm]:
```suggestion
	name = "improvised crowbar"
```

File: code/modules/species/species.dm [hunk 5]
```diff
@@ -647,29 +647,28 @@ The slots that you can use are found in items_clothing.dm and are the inventory
 	var/list/holding = list(target.get_active_hand() = 60, target.get_inactive_hand() = 30)
 
 	var/skill_mod = 10 * attacker.get_skill_difference(SKILL_COMBAT, target)
-	var/state_mod = attacker.melee_accuracy_mods() - target.melee_accuracy_mods()
-	var/stim_mod = target.chem_effects[CE_STIMULANT]
-	var/push_mod = min(max(1 + attacker.get_skill_difference(SKILL_COMBAT, target), 1), 3)
-	if(target.a_intent == I_HELP)
-		state_mod -= 30
+	var/state_mod = attacker.melee_accuracy_mods() - target.melee_accuracy_mods() + ((target.a_intent == I_HELP) ? 15 : 0)
```
Comment (EgorDinamit) [inline code/modules/species/species.dm:650]: I don't really get it. Why was the intent logic switched? What's going on in this file, in general?

**Review by Tsurupeta — CHANGES_REQUESTED** [review]
Make improvised tools as separate PR.

**Comment by aidan-bowman** [comment]
Changes have been made.

**Review by Tsurupeta — APPROVED** [review]

**Review by EgorDinamit — APPROVED** [review]
MERGING!

**Pull request merged**